apps/api/src/domain/models/bibliographic-reference/court-case-bibliographic-reference/commands/create-court-case-bibliographic-reference.command.integration.spec.ts

```diff
@@ -0,0 +1,185 @@
+
+const newUuidToUse = '9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dc246';
+
+const existingCourtCaseBibliographicReference = getValidBibliographicReferenceInstanceForTest(
+    BibliographicReferenceType.courtCase
+).clone({
+    id: dummyUuid,
+});
+
+const testDatabaseName = generateDatabaseNameForTestSuite();
+
+const buildValidCommandFSA = (
+    id: AggregateId
+): FluxStandardAction<DTO<CreateCourtCaseBibliographicReference>> => ({
+    type: commandType,
+    payload: {
+        id,
+        rawData: {
+            foo: 'hello world court case',
+        },
+        caseName: 'BC Supreme Court vs. Some Innocent Person',
+        abstract: 'The is the court case abstract.',
+        dateDecided: '2020-4-23',
+        court: 'BC Supreme Court',
+        url: 'https://bcsupremecourt.com',
+        pages: '47-78',
+    },
+});
+
+const dummyFSAFactory = new DummyCommandFSAFactory(buildValidCommandFSA);
+
+describe(`The command: ${commandType}`, () => {
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let idManager: IIdManager;
+
+    let assertionHelperDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager } = await setUpIntegrationTest({
+            ARANGO_DB_NAME: testDatabaseName,
+        }));
+
+        assertionHelperDependencies = {
+            testRepositoryProvider,
+            commandHandlerService,
+            idManager,
+        };
+
+        commandHandlerService.registerHandler(
+            commandType,
+            new CreateCourtCaseBibliographicReferenceCommandHandler(
+                testRepositoryProvider,
+                idManager
+            )
+        );
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    describe('when the command is valid', () => {
+        it('should succeed with appropriate updates to the database', async () => {
+            await assertCreateCommandSuccess(assertionHelperDependencies, {
+                buildValidCommandFSA,
+                initialState,
+                systemUserId: dummySystemUserId,
+                checkStateOnSuccess: async ({ id }: CreateCourtCaseBibliographicReference) => {
+                    await assertResourcePersistedProperly(idManager, testRepositoryProvider, {
+                        id,
+                        type: ResourceType.bibliographicReference,
+                    });
+                },
+            });
+        });
+    });
+
+    describe('when the command is invalid', () => {
+        describe('when the payload has an invalid type', () => {
+            generateCommandFuzzTestCases(CreateCourtCaseBibliographicReference).forEach(
+                ({ description, propertyName, invalidValue }) => {
```

Comment:
There's got to be some way to break this out into a util. I'll give it another try.